src/Util/log.cpp

```diff
@@ -7,8 +7,53 @@
 #include <vector>
 #include <Util/config.h>
 
+#ifdef _WIN64
+#include <Windows.h>
+#endif
+
+#include "log.h"
+
 namespace logging {
 std::vector<spdlog::sink_ptr> sinks;
+constexpr bool log_file_exceptions = true;  // disable it to disable logging
+
+void flush() { spdlog::details::registry::instance().flush_all(); }
+
+#ifdef _WIN64
+
+static LONG WINAPI exception_handler(PEXCEPTION_POINTERS pExp) noexcept {
+    const unsigned ec = pExp->ExceptionRecord->ExceptionCode;
+    switch (ec) {
+        case EXCEPTION_ACCESS_VIOLATION:
+            LOG_CRITICAL_IF(log_file_exceptions,"Exception EXCEPTION_ACCESS_VIOLATION ({}). ", log_hex(ec));
+            switch (pExp->ExceptionRecord->ExceptionInformation[0]) {
+                case 0: LOG_CRITICAL_IF(log_file_exceptions,"Read violation at address {}.", log_hex(pExp->ExceptionRecord->ExceptionInformation[1])); break;
+                case 1: LOG_CRITICAL_IF(log_file_exceptions,"Write violation at address {}.", log_hex(pExp->ExceptionRecord->ExceptionInformation[1])); break;
+                case 8:LOG_CRITICAL_IF(log_file_exceptions,"DEP violation at address {}.", log_hex(pExp->ExceptionRecord->ExceptionInformation[1])); break;
+                default: break;
+            }
+            break;
+        case EXCEPTION_ARRAY_BOUNDS_EXCEEDED: LOG_CRITICAL_IF(log_file_exceptions,"Exception EXCEPTION_ARRAY_BOUNDS_EXCEEDED ({}). ", log_hex(ec)); break;
```

Comment:
```suggestion
        case EXCEPTION_ARRAY_BOUNDS_EXCEEDED: LOG_CRITICAL_IF(log_file_exceptions,"Exception EXCEPTION_ARRAY_BOUNDS_EXCEEDED ({:#x}). ", ec); break;
```
fmt can format hex numbers without dedicated functions